src/Processors/Pipe.cpp

```diff
@@ -90,6 +90,10 @@ Pipe::Pipe(Pipes && pipes, ProcessorPtr transform)
 
         max_parallel_streams += pipe.max_parallel_streams;
         processors.insert(processors.end(), pipe.processors.begin(), pipe.processors.end());
+
+        std::move(pipe.table_locks.begin(), pipe.table_locks.end(), std::back_inserter(table_locks));
```

Comment:
The context of this code is not clear for me. Maybe add some comments?

Reply:
Comment is [here](https://github.com/ClickHouse/ClickHouse/blob/d79982f4973ccf8f0a9cc6198f23ec1f150494d6/src/Processors/Pipe.h#L85). These shared pointers were lost when creating a `Pipe` from other `Pipes`.

Reply:
Ok.